### ISSUE TITLE
(pedantry, feel free to ignore) Large files in repo

### ISSUE THREAD

**Issue by kdmurray91** [gbradburd/conStruct#11]
Hi Gideon,

I'm not sure if this was intentional, but it looks like there are still large files in the *git history* of the repo. Even thought the files no longer exist when checking out master, the whole repo (hidden history and all) is around 700mb. If you want to, you can run the following which will remove the `./data`, `./sims` and `./writeup` directories from all history, leaving only about 10mb of stuff in the code directory (which is now the root directory of the repo).  Git's a complete mindbender at times!

```
# Create tracking branches of all branches
for remote in `git branch -r | grep -v /HEAD`; do git checkout --track $remote ; done

git checkout master

# Remove the  $removeme directory from all commits, then remove the refs to the old commits
# (repeat these two commands for as many directories that you want to remove)
for removeme in data sims writeup
do
    git filter-branch --index-filter "git rm -rf --cached --ignore-unmatch ./$removeme" --prune-empty --tag-name-filter cat -- --all                                                                                                  
    git for-each-ref --format="%(refname)" refs/original/ | xargs -n 1 git update-ref -d
done

# Ensure all old refs are fully removed
rm -Rf .git/logs .git/refs/original

# Perform a garbage collection to remove commits with no refs
git gc --prune=all --aggressive
```

Then, after checking all's well, you can force-push to github.

I've run this on a local copy of the repo, and the history seems to have been preserved perfectly

Hope that's useful,
Kevin

**Comment by gbradburd** [comment]
Hi Kevin,

I think I've fixed it - I had actually filter-branch'd those directories previously, but I hadn't realized I had to force push against my tagged releases, which I think was the source of the problem(???).  Anyway, I just tried to clone the repo and the whole thing is <150Mb, so hopefully I've fixed the issue.  If I have, feel free to close the issue.  If that still didn't do the trick, let me know and I'll see what I can do...

Thanks for the input!
-g
